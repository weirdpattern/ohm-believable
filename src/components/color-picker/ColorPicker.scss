.color-picker {
  position: absolute;
  left: 50%;
  margin: 0 0 0 -100px
}
