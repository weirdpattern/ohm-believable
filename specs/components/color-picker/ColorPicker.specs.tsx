import * as React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";

import ColorPicker from "../../../src/components/color-picker";

describe("ColorPicker components", (): void => {
  it("renders correctly", (): void => {
    const colors: string[] = [
      "#000000",
      "#FF0000",
      "#FFFF00",
      "#FFFFFF",
      "#0000FF",
      "#00FFFF"
    ];

    const tree: ReactTestRendererJSON = renderer
      .create(
        <ColorPicker
          colors={colors}
          onColorChange={null}
          onEditBandTimeout={null}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
