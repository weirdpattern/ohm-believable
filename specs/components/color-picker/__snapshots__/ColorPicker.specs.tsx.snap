// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker components renders correctly 1`] = `
<div
  className="color-picker"
>
  <div
    className="github-picker "
    style={
      Object {
        "MozBorderRadius": "4px",
        "MozBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
        "OBorderRadius": "4px",
        "OBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
        "WebkitBorderRadius": "4px",
        "WebkitBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
        "background": "#fff",
        "border": "1px solid rgba(0,0,0,0.2)",
        "borderRadius": "4px",
        "boxShadow": "0 3px 12px rgba(0,0,0,0.15)",
        "display": "flex",
        "flexWrap": "wrap",
        "msBorderRadius": "4px",
        "msBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
        "padding": "5px",
        "position": "relative",
        "width": "200px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "8px solid transparent",
          "borderBottomColor": "rgba(0,0,0,0.15)",
          "display": "none",
          "position": "absolute",
        }
      }
    />
    <div
      style={
        Object {
          "border": "7px solid transparent",
          "borderBottomColor": "#fff",
          "display": "none",
          "position": "absolute",
        }
      }
    />
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#000000",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#000000"
          />
        </span>
      </div>
    </span>
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#FF0000",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#FF0000"
          />
        </span>
      </div>
    </span>
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#FFFF00",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#FFFF00"
          />
        </span>
      </div>
    </span>
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#FFFFFF",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#FFFFFF"
          />
        </span>
      </div>
    </span>
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#0000FF",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#0000FF"
          />
        </span>
      </div>
    </span>
    <span
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        style={
          Object {
            "fontSize": "0",
            "height": "25px",
            "width": "25px",
          }
        }
      >
        <span
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              Object {
                "background": "#00FFFF",
                "cursor": "pointer",
                "height": "100%",
                "outline": "none",
                "position": "relative",
                "width": "100%",
              }
            }
            tabIndex={0}
            title="#00FFFF"
          />
        </span>
      </div>
    </span>
  </div>
</div>
`;
